### ISSUE TITLE
Installation failure on Ubuntu 20.04LTS Vitrualbox VM

### ISSUE THREAD

**Issue by KDubya17** [vincentcasseau/hyStrath#105]
I am using Ubuntu 20.04LTS VM to practice the installation before doing it on a physical machine.  I have been following the guide on the hystrath.github.io.  I am struggling with the last step.  Note in Step #4 the following packages/libraries are not available for Ubuntu 20.04LTS:

` sudo apt-get install qt4-dev-tools libqt4-dev libqt4-opengl-dev freeglut3-dev libqtwebkit-dev`

so, I installed the Qt tools for Unbuntu 20.04LTS i.e. qtcreator and qt5-default.  But I doubt that could be the cause of the error,  but I could be wrong.  Also prior to Step #5, I add the following to my ~/.bashrc:

'source $HOME/OpenFOAM/OpenFOAM-v1706/etc/bashrc'

At Step #5, I do get a error when doing ./Allwmake.  Using foamInstallationTest, it says that icoFoam is not installed.  From the CFD-Online forum, some have had success using ./Allwmake -j  -s -l to fix that error, but for me the error still persists.  However, I proceeded with the last step anyways:

`./install.sh 6 2>/dev/null`

I attached the installation logs for the modules.  I am really interested in the dsmc, but I was thinking of dabbling with the other modules.

[of1706_logInstall.zip](https://github.com/vincentcasseau/hyStrath/files/8996039/of1706_logInstall.zip)

In short, wmkdep is not finding the header files for the compilation.  After trying a few different things, I was able to get a different error of the wmkdep command no being found, but I have not been able to recreate it. Let me know of any other information you may need.

Thank You.
#

**Comment by KDubya17** [comment]
I followed your suggestion for Step 4. The packages/libraries installed. I verified the build folder has the six bash files. I was running install.sh from the htStrath directory. Despite repeating Step 5, icoFoam solver is not installed correctly. When running ./Allwmake in Step 5, I notice the compilers returns the following:

```
/home/kyle/OpenFOAM/OpenFOAM-v1706/src/OpenFOAM/lnInclude/PackedListI.H:549:13: error: return-statement with a value, in function returning ‘void’ [-fpermissive]
  549 |     return *this;
```

and

```
make: *** [/home/kyle/OpenFOAM/OpenFOAM-v1706/wmake/rules/General/transform:29: /home/kyle/OpenFOAM/OpenFOAM-v1706/build/linux64GccDPInt32Opt/src/OSspecific/POSIX/fileMonitor.o] Error 1
```

Here are the recent install logs:

[installation-logs_220628.zip](https://github.com/vincentcasseau/hyStrath/files/9006078/installation-logs_220628.zip)

**Comment by vincentcasseau** [comment]
icoFoam won't run and hyStrath won't be installed until the Allwmake script in Step 5 returns at least one error.
Could you please send me the log of Allwmake?
What do you get when typing (see step 1):

```sh
gcc --version
```

**Comment by KDubya17** [comment]
The current gcc version was 9.4.0.  I reran:

```
sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-7 7
sudo update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-7 7
sudo update-alternatives --config gcc
sudo update-alternatives --config g++
```
to get gcc 7.  When using update-alternatives --install, the terminal mentions a force install due to a broken link. After that, OpenFOAM compiled correctly and icoFoam was installed. I did get an error installing the cfd-module when trying to install all the modules at once. Looking at the cfd-module installation log:

[installation_logs_220629.zip](https://github.com/vincentcasseau/hyStrath/files/9014904/installation_logs_220629.zip)
e attached log) 

I notice it was when trying to look for some dsmc files, so thinking it needed the dsmc-module installed, I ran the cfd-module installation again alone and no errors in the installation log of the cfd-module.  I am wondering if some of the packages/libraries undid the initial downgrade.

**Comment by vincentcasseau** [comment]
You did the right by relaunching the compilation.
I fixed this issue in commit 2ede579.
